pmlc/dialect/tile/builder.h

```diff
@@ -107,7 +107,15 @@ class TileBuilder {
   void SetUseDefault(mlir::Value cion, mlir::Value defaultValue);
   void SetNoReduce(mlir::Value cion, bool no_reduce);
 
-  std::shared_ptr<TileProgram> MakeProgram(llvm::StringRef name, const ProgramMutations& mutations);
+  std::shared_ptr<TileProgram> MakeProgram(llvm::StringRef name, const ProgramMutations& mutations,
+                                           util::DataType floatx = DataType::invalid,
+                                           util::DataType intx = DataType::invalid);
+
+  std::shared_ptr<TileProgram> MakeProgram(llvm::StringRef name, const ProgramMutations& mutations,
+                                           std::uint64_t floatx, std::uint64_t intx) {
+    return MakeProgram(name, mutations, util::symbolizeDataType(floatx).getValueOr(DataType::invalid),
+                       util::symbolizeDataType(intx).getValueOr(DataType::invalid));
+  }
```

Comment:
Let's isolate the conversions between raw integers to strongly typed enums in the FFI, so this method should be dropped.